databroker/tests/test_document.py

```diff
@@ -48,3 +50,24 @@ def test_to_dict():
     b['x']['y']['z'] = 2
     # Verify original is not modified.
     assert a['x']['y']['z'] == 1
+
+
+def test_pickle_round_trip():
+    expected = Document({'x': {'y': {'z': 1}}})
+    serialized = pickle.dumps(expected)
+    actual = pickle.loads(serialized)
```

Comment:
Why not to use the same pattern (one call for dumping and loading) as in the 2 tests below?

Reply:
This test, which was the first commit in the PR and precedes the associated fix, is meant to clearly illustrate the _pickling_ succeeds but _unpickling_ raises an error. By separating them into two lines, it's easier to understand where the failure is.

I imagine that Tom decided to do JSON and msgpack in one line because those currently work.

Reply:
Indeed, I developed the later tests by typing into a terminal and then lazily copy-pasted the code into the test.